src/cpp/database/database.hpp

```diff
@@ -412,12 +412,6 @@ class Database
             throw BadParameter("Endpoint GUID cannot be empty");
         }
 
-        /* Check that locators is not empty */
-        if (endpoint->locators.empty())
-        {
-            throw BadParameter("Endpoint locators cannot be empty");
-        }
-
```

Comment:
We should keep these lines. Whereas the locator can be created without knowing to which endpoint is related, the endpoint always knows which locators are related to itself.